src/test/java/TraineeServiceTest.java

```diff
@@ -0,0 +1,95 @@
+import org.example.dao.TraineeDao;
+import org.example.models.Trainee;
+import org.example.service.TrainingService;
+import org.example.service.impl.TraineeServiceImpl;
+import org.example.util.Generator;
+import org.example.util.UserUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class TraineeServiceTest {
+
+    @Mock
+    private TraineeDao traineeDao;
+
+    @Mock
+    private UserUtil userUtil;
+    @Mock
+    private TrainingService trainingService;
+
+    @InjectMocks
+    private TraineeServiceImpl traineeService;
+
+    @BeforeEach
```

Comment:
That looks strange setting it before each test. Why was it done?

Reply:
When I invoke createTrainee method form traineeServiceImpl class, it uses userService's getAllExistingUsernames method, but userService is null. So I "inject it manually"